service/storage/operations_internal_test.go

```diff
@@ -1255,3 +1256,85 @@ func TestIndexAndLookup_Seals(t *testing.T) {
 		assert.Equal(t, 1, decodeCallCount)
 	})
 }
+
+func TestLibrary_IterateLedger(t *testing.T) {
```

Comment:
This test case does not test:

- whether it correctly skips the older payloads when multiple payloads that have the same path; or
- whether it uses the correct path and payload in the callback.